File: dart/lib/model/util_viewer_response.dart
part of swagger.api;

class UtilViewerResponse {
  
  String address = null;

  String authExpiry = null;

  int id = null;

  List<String> miners = [];

  int perms = null;

  UtilUserSettings settings = null;

  String username = null;

  UtilViewerResponse();

  @override
  String toString() {
    return 'UtilViewerResponse[address=$address, authExpiry=$authExpiry, id=$id, miners=$miners, perms=$perms, settings=$settings, username=$username, ]';
  }

  UtilViewerResponse.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    address = json['address'];
    authExpiry = json['auth_expiry'];
    id = json['id'];
    miners = (json['miners'] as List).map((item) => item as String).toList();
    perms = json['perms'];
    settings = new UtilUserSettings.fromJson(json['settings']);
    username = json['username'];
  }

  Map<String, dynamic> toJson() {
    return {
      'address': address,
      'auth_expiry': authExpiry,
      'id': id,
      'miners': miners,
      'perms': perms,
      'settings': settings,
      'username': username
     };
  }

  static List<UtilViewerResponse> listFromJson(List<dynamic> json) {
    return json == null ? new List<UtilViewerResponse>() : json.map((value) => new UtilViewerResponse.fromJson(value)).toList();
  }

  static Map<String, UtilViewerResponse> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, UtilViewerResponse>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new UtilViewerResponse.fromJson(value));
    }
    return map;
  }
}

File: csharp-dotnet2/src/main/CsharpDotNet2/estuary-client/Model/UtilViewerResponse.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class UtilViewerResponse {
    /// <summary>
    /// Gets or Sets Address
    /// </summary>
    [DataMember(Name="address", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "address")]
    public string Address { get; set; }

    /// <summary>
    /// Gets or Sets AuthExpiry
    /// </summary>
    [DataMember(Name="auth_expiry", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "auth_expiry")]
    public string AuthExpiry { get; set; }

    /// <summary>
    /// Gets or Sets Id
    /// </summary>
    [DataMember(Name="id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "id")]
    public int? Id { get; set; }

    /// <summary>
    /// Gets or Sets Miners
    /// </summary>
    [DataMember(Name="miners", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "miners")]
    public List<string> Miners { get; set; }

    /// <summary>
    /// Gets or Sets Perms
    /// </summary>
    [DataMember(Name="perms", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "perms")]
    public int? Perms { get; set; }

    /// <summary>
    /// Gets or Sets Settings
    /// </summary>
    [DataMember(Name="settings", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "settings")]
    public UtilUserSettings Settings { get; set; }

    /// <summary>
    /// Gets or Sets Username
    /// </summary>
    [DataMember(Name="username", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "username")]
    public string Username { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class UtilViewerResponse {\n");
      sb.Append("  Address: ").Append(Address).Append("\n");
      sb.Append("  AuthExpiry: ").Append(AuthExpiry).Append("\n");
      sb.Append("  Id: ").Append(Id).Append("\n");
      sb.Append("  Miners: ").Append(Miners).Append("\n");
      sb.Append("  Perms: ").Append(Perms).Append("\n");
      sb.Append("  Settings: ").Append(Settings).Append("\n");
      sb.Append("  Username: ").Append(Username).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
